src/entities/repositories/repository.service.ts

```diff
@@ -64,4 +77,233 @@ export class RepositoryService {
     }
     return repo;
   }
+
+  public preAggregate(
+    filter: FilterQuery<RepositoryDocument>,
+    options: {
+      issues?: {
+        labels?: boolean;
+        assignees?: boolean;
+        events?: {
+          actor?: boolean;
+          since?: string;
+          to?: string;
+        };
+        since?: string;
+        to?: string;
+      };
+      commits?: {
+        author?: boolean;
+        since?: string;
+        to?: string;
+      };
+      releases?: {
+        since?: string;
+        to?: string;
+      };
+      diffs?: boolean;
+    },
+  ): Aggregate<any[]> {
```

Comment:
I would have to look at it, but I guess it would be possible.

Reply:
Sadly, it's not. If the prepared pipeline was directly executed, it would be no problem. But since we have the clear intention to add additional stages to the pipeline, the resulting type changes.